spec/models/traversable_spec.rb

```diff
@@ -0,0 +1,93 @@
+RSpec.describe Traversable do
+  subject(:traversable) do
+    described_class.new(service: service, flow_uuid: flow_uuid)
+  end
+
+  describe '#question_pages' do
+    context 'simple flow with no branching' do
+      let(:metadata) { metadata_fixture(:version) }
+      let(:service) { MetadataPresenter::Service.new(metadata) }
+
+      context 'adding a branch at the end of the flow' do
+        let(:flow_uuid) { service.pages[-3].uuid } # page before check answers
+        let(:expected_urls) do
+          %w[
+            name
+            email-address
+            parent-name
+            your-age
+            family-hobbies
+            do-you-like-star-wars
+            holiday
+            burgers
+            star-wars-knowledge
+            dog-picture
+          ]
+        end
+
+        it 'returns the traversable question pages' do
+          traversable_urls = traversable.question_pages.map(&:url)
+          expect(traversable_urls).to eq(expected_urls)
+        end
+      end
+
+      context 'adding a branch in the middle of the flow' do
+        let(:flow_uuid) { service.pages[3].uuid }
+        let(:expected_urls) { %w[name email-address parent-name] }
+
+        it 'returns the traversable question pages' do
+          traversable_urls = traversable.question_pages.map(&:url)
+          expect(traversable_urls).to eq(expected_urls)
+        end
+      end
+    end
+
+    context 'simple branching flow' do
+      let(:metadata) { metadata_fixture(:branching) }
+      let(:service) { MetadataPresenter::Service.new(metadata) }
+
+      context 'adding a branch at the end of the flow' do
+        let(:flow_uuid) { service.pages[-3].uuid } # page before check answers
+        let(:expected_urls) do
+          %w[name
+             do-you-like-star-wars
+             star-wars-knowledge
+             favourite-fruit
+             apple-juice
+             orange-juice
+             favourite-band
+             music-app
+             best-formbuilder
+             which-formbuilder
+             burgers
+             marvel-series
+             best-arnold-quote]
+        end
+
+        it 'returns the traversable question pages' do
+          traversable_urls = traversable.question_pages.map(&:url)
+          expect(traversable_urls).to eq(expected_urls)
+        end
+      end
+
+      context 'adding a branch in the middle of the flow' do
+        let(:flow_uuid) { service.pages[8].uuid } # music-app (currently)
+        let(:expected_urls) do
+          %w[name
+             do-you-like-star-wars
+             star-wars-knowledge
+             favourite-fruit
+             apple-juice
+             orange-juice
+             favourite-band
+             music-app]
+        end
+
+        it 'returns the traversable question pages' do
+          traversable_urls = traversable.question_pages.map(&:url)
+          expect(traversable_urls).to eq(expected_urls)
+        end
+      end
+    end
+  end
+end
```

Comment:
Can. you add a scenario where there are no next pages?

A -> B -> C -> (none)

where C is the flow_uuid?

Or add a scenario where A is the flow uuid?

Or B and return empty because start page doesn't have a questioon?

Reply:
when there are no next pages default_next will be any empty string so the loop should stop. confirmation pages are like this, and we should make sure exit pages are the same too. if we start supporting adding branches without pointing to a destination this should also be covered.

will add test for no question pages

also want to test the route model specifically. 

i am torn as to whether to add more complex branching at this point or to wait until the editor is more complete